mobilizon_bots/main.py

```diff
@@ -37,14 +45,29 @@ async def main(settings_file):
     unpublished_events = get_unpublished_events(published_events)
     # Store in the DB only the ones we didn't know about
     await create_unpublished_events(unpublished_events, active_publishers)
-    event = select_event_to_publish(published_events, unpublished_events)
+    event = select_event_to_publish(
+        published_events,
+        # We must load unpublished events from DB since it contains
+        # merged state between Mobilizon and previous WAITING events.
+        list(await get_db_unpublished_events()),
+    )
     if event:
         logger.debug(f"Event to publish found: {event.name}")
-        result = PublisherCoordinator(event).run()
+        report = PublisherCoordinator(
+            event,
+            [
+                (pub.id, pub.publisher.name)
+                for pub in await publications_with_status(
+                    status=PublicationStatus.WAITING,
+                    event_mobilizon_id=event.mobilizon_id,
+                )
+            ],
+        ).run()
+        await save_publication_report(report, event)
 
         logger.debug("Closing")
 
-        await graceful_exit(0 if result.successful else 1)
+        exit(0 if report.successful else 1)
```

Comment:
is this on purpose?

Reply:
Actually no, I think it must have slipped in. Now is fixed